service/src/main/java/bio/terra/tanagra/service/export/impl/IndividualFileDownload.java

```diff
@@ -44,10 +45,18 @@ public void initialize(DeploymentConfig deploymentConfig) {
 
   @Override
   public ExportResult run(ExportRequest request) {
+    String cohortRef =
+        simplifyStringForName(
+                request.getCohorts().get(0).getDisplayName()
+                    + "_"
+                    + request.getCohorts().get(0).getId())
+            + (request.getCohorts().size() > 1
+                ? "plus" + (request.getCohorts().size() - 1) + "more"
+                : "");
     Map<String, String> entityToGcsUrl =
-        request.writeEntityDataToGcs("tanagra_${entity}_" + Instant.now());
+        request.writeEntityDataToGcs("${entity}_cohort" + cohortRef + "_${random}");
     Map<Cohort, String> cohortToGcsUrl =
-        request.writeAnnotationDataToGcs("tanagra_${cohort}_" + Instant.now());
+        request.writeAnnotationDataToGcs("annotations_cohort${cohort}_${random}");
 
```

Comment:
including study_id also would be better, IMHO. Similar to the IpynbFileDownload.java#71 below

Reply:
If anything, I think the underlay would actually be the most valuable to add since it seems a lot more likely to export the same data/cohorts on different data releases than entirely different studies

Reply:
Easy enough to include both studyId and underlayName. We may have to pare this down in the future if the names get too long, but we can include everything at first.

I guess this is more of a product question, but I'm skeptical about the utility of the ids (cohort & study). They're not visible anywhere in the UI, so seems unlikely to be useful to the user. And the main purpose of having a readable filename here is so the user can make sense of their downloads folder. Secondary benefit is helping with debugging, but the random suffix would be more reliable for that anyway. We can bring up in the weekly meeting, as a low priority change for now.

Reply:
RE ipynb file download: That line is actually writing the study name/id into the ipynb file, not as the filename.